src/Common/ExecCommand.php

```diff
@@ -233,13 +254,15 @@ protected function executeCommand($command)
             $process->setWorkingDirectory($this->workingDirectory);
         }
         $this->getExecTimer()->start();
-        if ($this->isPrinted) {
-            $process->run(function ($type, $buffer) {
-                print $buffer;
-            });
-        } else {
-            $process->run();
-        }
+        $process->run(
+            function ($type, $buffer) {
+                if ($this->isOutputLogged) {
+                    $this->printTaskInfo($buffer);
+                } elseif ($this->isOutputPrinted) {
+                    print($buffer);
```

Comment:
It seems useful to be able to set the log level at which `exec` output is printed (e.g. if calling `exec(git merge);` on a the merge of a large build result set, perhaps you want to see that output only under -vvv mode). However, I am not convinced that we want to go so far as to add this same flexibility to all tasks, or to make the various changes to logging proposed here. If we could limit this PR to just the minimum changes needed to associate a log level with printing the output buffer, then I'd be in support of adding it.